specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2019-06-04/imds.json

```diff
@@ -0,0 +1,828 @@
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    }
+                },
+                "x-ms-examples": {
+                    "Get Instance Metadata for the VM": {
+                        "$ref": "./examples/GetInstanceMetadata.json"
+                    }
+                }
+            }
+        },
+        "/attested/document": {
+            "get": {
+                "operationId": "Attested_GetDocument",
+                "description": "Get Attested Data for the Virtual Machine.",
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "name": "nonce",
+                        "in": "query",
+                        "required": false,
+                        "type": "string",
+                        "description": "This is a string of up to 32 random alphanumeric characters."
+                    },
+                    {
+                        "$ref": "#/parameters/MetadataParameter"
+                    } 
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK",
+                        "schema": {
+                            "$ref": "#/definitions/AttestedData"
+                        }
+                    },
+                    "default": {
+                        "description": "Error response describing why the operation failed.",
+                        "schema": {
+                            "$ref": "#/definitions/ErrorResponse"
+                        }
+                    }
+                },
+                "x-ms-examples": {
+                    "Get Attested Data for the VM": {
+                        "$ref": "./examples/GetAttestedData.json"
+                    }
+                }
+            }
+        },
+        "/identity/oauth2/token": {
+            "get": {
+                "tags": [
+                    "Get token"
+                ],
+                "description": "Get a Token from Azure AD",
+                "operationId": "Identity_GetToken",
+                "produces": [
+                    "application/json"
+                ],
+                "parameters": [
+                    {
+                        "$ref": "#/parameters/MetadataParameter"
+                    },
+                    {
+                        "name": "resource",
+                        "in": "query",
+                        "type": "string",
+                        "description": "This is the urlencoded identifier URI of the sink resource for the requested Azure AD token. The resulting token contains the corresponding aud for this resource.",
+                        "required": true
```

Comment:
Why this parameter is required in query? What will happen when users don't provide `resource` parameter?

Reply:
The token returned by this query is specific to a resource. Without this parameter, the query will fail.